rfd/0036/README.md

```diff
@@ -13,281 +13,800 @@ state: draft
+- This demonstrates colon-separated/joined naming scheme, as in `group:resource:<tag name>`, but `group.resource.<tag name>` makes about equal sense to the author
+- Actually, this example begs for uniform resource naming within Triton
+
+
+
+### Auditing
+
+Automation without auditability is dangerous. [RFD50](https://github.com/joyent/rfd/blob/master/rfd/0050/README.md) focuses on improving audit logs for individual instances, but we must recognize the need to audit the resources and automation proposed in this RFD. These logs will be critical for both data center operators and end users. Events that must be auditable include create, update, and delete operations on a service, the automatic scheduling and rescheduling of instances, as well as failed health checks and other events.
+
+
+
+## Strawperson examples
+
+These are being rewritten from examples of the service manifest and Triton CLI commands offered in previous versions of the RFD.
+
+Previous examples:
+
+- [Service manifest](https://github.com/joyent/rfd/blob/e4a66d6b5754a045502f971deaedef1c8b8be138/rfd/0036/services/manifest.md)
+- [Managing services (via projects) in Triton CLI](https://github.com/joyent/rfd/blob/e4a66d6b5754a045502f971deaedef1c8b8be138/rfd/0036/projects/triton-projects-cli.md)
+- [Metadata and variable interpolation in the service manifest](https://github.com/joyent/rfd/tree/e4a66d6b5754a045502f971deaedef1c8b8be138/rfd/0036/meta)
+
+*NEEDS REVISION*
+
+
+
+## Desirable features not addressed in this RFD
+
+This section discusses features that might make it easier to operate [stateful applications that require external state management](#stateful-with-external-state-management), as well as one that would improve the operation of [applications presented as services to others](#applications-presented-as-services-to-others). These features are not being further addressed in this RFD, but this section summarizes them as they relate to this RFD.
+
+Also, because this RFD assumes the ability to define services running Docker images, all without interaction with the Docker API presented by sdc-docker, we should consider CloudAPI features to support Docker API-style interactions with instances, such as some equivalent of `docker exec`.
+
+
+
+### Reprovisioning via public APIs
+
+Though [`vmadam`](https://wiki.smartos.org/display/DOC/Using+vmadm+to+manage+virtual+machines#Usingvmadmtomanagevirtualmachines-UpdatingaVM) and the [operator portal](https://docs.joyent.com/private-cloud/instances) support reprovisioning instances, the feature is [not exposed publicly via CloudAPI](https://github.com/joyent/node-triton/issues/141). Triton and Manta upgrades are done via this private reprovisioning feature. The instance's primary volume is overwritten with a new image while stateful data in the instance is preserved on another volume in the dataset. This process depends on privileged access, either with a delegated dataset in the instance or with privileges in the global zone
```

Comment:
This stands in contradiction to the traditional DBA wisdom that your call to a database should *never* go through DNS.